### ISSUE TITLE
GitHub Metadata : build_revision is always NIL 

### ISSUE THREAD

**Issue by BenSouchet** [pages-themes/cayman#145]
### TL;DR :
I have currently an issue with themes for github pages where the variable `site.github.build_revision` is always NIL.
I think this it's an issue when the github page is build, the variable git_commit_sha cannot be retrieve.

### Informations
To be sure it's global I did some basic reproduction steps:
- Creating a new github repository
- Added a README.md
- Configure a Github page in the Github repository settings
- Selected a Theme (for exemple **Cayman**)
- Waited the end of the build in the Actions tab
- Locked in the page source of the github.io page for the stylesheet link where the version need to be concatenated

The link is `<link rel="stylesheet" href="/test_101/assets/css/style.css?v=">`, there is nothing after the `v=`.

I checked every GitHub Metadata by adding these lines in the html :
```
<pre id="jekyll-debug"></pre>
<script>
  var obj = JSON.parse(decodeURIComponent("{{ site.github | jsonify | uri_escape }}"));
  var prettyJson = JSON.stringify(obj, null, 4);  // Pretty-printed JSON (indented 4 spaces).
  document.getElementById("jekyll-debug").textContent = prettyJson;
</script>
```
and the **build_revision** (a.k.a git_commit_sha) is the only empty value.

### More details
- At the end of the "Build page with Jekyll" part of the build process we can found that:
```GitHub Metadata: Generating for BenSouchet/test_101
   GitHub Metadata: Calling @client.repository("BenSouchet/test_101", {:accept=>"application/vnd.github.drax-preview+json"})
   GitHub Metadata: Calling @client.pages("BenSouchet/test_101", {})
fatal: not a git repository (or any of the parent directories): .git
           Writing: /github/workspace/docs/_site/assets/css/style.css
           Writing: /github/workspace/docs/_site/index.html
                    done in 0.624 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
```
**So there is an ISSUE** , maybe related to [this](https://github.com/rubygems/bundler/issues/6453).
- I tried to add lot of settings in the `_config.yml` my file ended up a bit like that:
```
theme: jekyll-theme-cayman
title: Test 101
description: Bug testing
repository: BenSouchet/test_101
encoding: UTF-8
plugins:
- jekyll-remote-theme
- jekyll-github-metadata
```

@benbalter I think you are the man of the situation, you will I think find the bug in no time 🥇

**Comment by BenSouchet** [comment]
Good news the issue will be fixed soon ( https://github.com/jekyll/github-metadata/issues/220#issuecomment-1006765976 ), I close this issue, new comments will be on the issue I linked.
